Code/GraphMol/SynthonSpaceSearch/CMakeLists.txt

```
@@ -2,17 +2,23 @@
 rdkit_library(SynthonSpaceSearch
         SynthonSpaceSearch_details.cpp SynthonSpace.cpp SynthonSet.cpp Synthon.cpp
         SynthonSpaceSearcher.cpp SynthonSpaceSubstructureSearcher.cpp SynthonSpaceFingerprintSearcher.cpp
-        LINK_LIBRARIES SmilesParse FileParsers ChemTransforms Fingerprints SubstructMatch GraphMol)
+        SynthonSpaceRascalSearcher.cpp SynthonSpaceHitSet.cpp SearchResults.cpp MemoryMappedFileReader.cpp
+        LINK_LIBRARIES SmilesParse FileParsers ChemTransforms Fingerprints SubstructMatch GraphMol RascalMCES
+        GeneralizedSubstruct)
 target_compile_definitions(SynthonSpaceSearch PRIVATE RDKIT_SYNTHONSPACESEARCH_BUILD)
 
 rdkit_headers(SynthonSpace.h SynthonSet.h Synthon.h SearchResults.h
         SynthonSpaceSearcher.h SynthonSpaceSubstructureSearcher.h SynthonSpaceFingerprintSearcher.h
-        DEST GraphMol/SynthonSpaceSearch)
+        SynthonSpaceRascalSearcher.h DEST GraphMol/SynthonSpaceSearch)
 
 rdkit_catch_test(testSynthonSpaceSubstructureSearch substructure_search_catch_tests.cpp
         LINK_LIBRARIES SubstructMatch SubstructLibrary SynthonSpaceSearch)
 rdkit_catch_test(testSynthonSpaceFingerprintSearch fingerprint_search_catch_tests.cpp
         LINK_LIBRARIES SubstructMatch SubstructLibrary SynthonSpaceSearch)
+rdkit_catch_test(testSynthonSpaceRascalSearch rascal_search_catch_tests.cpp
+        LINK_LIBRARIES SubstructMatch SubstructLibrary SynthonSpaceSearch)
+rdkit_catch_test(testSynthonSpaceLONG_TEST LONG_TEST_catch_tests.cpp
```

I really like the idea of making the tests run faster!
I'd suggest that we disable these tests by default:
```suggestion
if(BUILD_LONG_RUNNING_TESTS)
        rdkit_catch_test(testSynthonSpaceLONG_TEST LONG_TEST_catch_tests.cpp
                LINK_LIBRARIES SubstructMatch SubstructLibrary SynthonSpaceSearch)
endif()
```

Then doing `cmake -DBUILD_LONG_RUNNING_TESTS=ON` will enable these tests and they can be disabled again with `cmake -DBUILD_LONG_RUNNING_TESTS=OFF`

---

Should the CI builds always run them?  In which case, I'll add a default ON to the main CMakeLists.txt.

---

> Should the CI builds always run them? In which case, I'll add a default ON to the main CMakeLists.txt.

I don't think the CI builds should run them. Those machines are too slow and our CI builds take long enough already.
We should be sure that all functionality is covered in the normal tests and that these just cover more extensive testing or testing with larger data.